blackjack/lib/blackjack/game.ex

```diff
@@ -0,0 +1,162 @@
+defmodule Blackjack.Game do
+
+  # New game method
+  def new_game() do
+    %Blackjack.State{}
+      |> Map.put(:deck_state, Deck.shuffle())
+  end
+
+  # State  
+  def check(game) do
+  %{
+    game_state: game.game_state,
+    deck_state: game.deck_state,
+    dealer_hand: game.dealer_hand,
+    dealer_value: game.dealer_value,
+    player_hand: game.player_hand,
+    player_value: game.player_value,
+    turn: game.turn
+  }
+  end
+
+  # New game setup
+  def hit(game = %{ game_state: :init }) do
+    {[card1, card2, card3, card4], new_deck} = Deck.deal(game.deck_state, 4)
+    game
+    |> Map.put(:game_state, :ongoing)
+    |> Map.put(:player_hand, [card1, card3])
+    |> Map.put(:dealer_hand, [card2, card4])
+    |> Map.put(:deck_state, new_deck)
+    |> count_hands
+    |> return_check
+  end
+
+  # Dealer hit
+  def hit(game = %{ turn: :dealer }) do
+    dealer(game, game.dealer_value, game.player_value)
+  end
+
+  # Player hit
+  def hit(game = %{ turn: :player }) do
+    {card, new_deck} = Deck.deal(game.deck_state, 1)
+    game
+    |> Map.put(:player_hand, card ++ game.player_hand)
+    |> Map.put(:deck_state, new_deck)
+    |> count_hands
+    |> bust_check(game.player_value)
+    |> return_check  
+  end
+
+  #Player stand
+  def stand(game = %{ turn: :player }) do
+    game
+    |> Map.put(:turn, :dealer)
+    |> return_check
+  end
+
+  ######## Private Methods ########
+  
+  # Returns game state
+  defp return_check(game) do
+    { game, check(game) }
+  end
+
+  # Dealer helper function
+  defp dealer(game, dealer_val, player_val) do
+    case dealer_val do
```

Comment:
and another case that should be a cond... (I read this file backwards, so see below)